internal/serialization/default_serializer.go

```diff
@@ -110,70 +113,80 @@ func (*BoolSerializer) Write(output serialization.DataOutput, i interface{}) {
 	output.WriteBool(i.(bool))
 }
 
-type UInteger16Serializer struct{}
+type UInt16Serializer struct{}
 
-func (*UInteger16Serializer) ID() int32 {
-	return ConstantTypeUInteger16
+func (*UInt16Serializer) ID() int32 {
+	return TypeUInt16
 }
 
-func (*UInteger16Serializer) Read(input serialization.DataInput) interface{} {
+func (*UInt16Serializer) Read(input serialization.DataInput) interface{} {
 	return input.ReadUInt16()
 }
 
-func (*UInteger16Serializer) Write(output serialization.DataOutput, i interface{}) {
+func (*UInt16Serializer) Write(output serialization.DataOutput, i interface{}) {
 	output.WriteUInt16(i.(uint16))
 }
 
-type Integer16Serializer struct{}
+type IntSerializer struct{}
 
-func (*Integer16Serializer) ID() int32 {
-	return ConstantTypeInteger16
+func (*IntSerializer) ID() int32 {
+	return TypeInt64
```

Comment:
Unfortunately there's no way around it. Go's `int` type is an alias to `int32` on 32-bit platforms but it's `int64` on 64-bit platforms. Since the platform used for serialization and deserialization may be different, and we don't have an integer type which can be either 32 or 64 bits in the serialization system, we have to choose between `int32` and `int64`. If we choose the former, we should panic if the user tries to set a value > `MaxInt32`. Using `int64` seems like the easiest and safest way.